opentracing-spring-web/src/main/java/io/opentracing/contrib/spring/web/webfilter/TracingOperator.java

```diff
@@ -72,7 +76,21 @@ public void subscribe(final CoreSubscriber<? super Void> subscriber) {
 
         try (final Scope scope = tracer.scopeManager().activate(span, false)) {
             exchange.getAttributes().put(TracingWebFilter.SERVER_SPAN_CONTEXT, span.context());
-            source.subscribe(new TracingSubscriber(subscriber, exchange, context, span, spanDecorators));
+            source.subscribe(new TracingSubscriber(subscriber, this.wrappleExchange(span), context, span, spanDecorators));
         }
     }
+
+    private ServerWebExchange wrappleExchange(Span span) {
+        Map<String, String> map = new HashMap();
+        this.tracer.inject(span.context(), Format.Builtin.HTTP_HEADERS, new TextMapAdapter(map));
+        ServerHttpRequest req = this.exchange.getRequest();
+        Iterator iterator = map.keySet().iterator();
+
+        while (iterator.hasNext()) {
```

Comment:
Since this is an iterator, would not be better using a "simple" `for (String key : map.keySet())`? This way you don't have to manually call `iterator.next()`.

Reply:
yes, thank you